src/logic/calculate.js

```diff
@@ -0,0 +1,15 @@
+import operate from './operate';
+
+const calculate = (data, button) => {
+  const returnData = data;
+  if (button === '+/-') {
+    returnData.total *= -1;
+    returnData.next *= -1;
+  } else {
+    returnData.total = operate(data.total, data.next, data.operation);
+  }
+
+  return returnData;
+};
```

Comment:
This logic is not comprehensive enough.

Think of cases you need to handle, please remember that `calculate` is the only place where the app logic would be. For each click, `calculate` function will be called.

Think of this and make changes accordingly.